### PR TITLE
feat(runtime): add `enable-lock` for swcli build and refactor runtime build/restore process with runtime.yaml 2.0 definition

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
- `swcli runtime build` does not depend on the current shell environment.
- `swcli runtime build` process is explicit in that it only uses `runtime.yaml` to build swrt. 
- Simplify to generate venv.tar in local_gen mode.
- related issues: 
  - https://github.com/star-whale/starwhale/issues/741
  - https://github.com/star-whale/starwhale/issues/765

## Feature Show
- no shell environment dependency
image
- venv-local-gen/restore

image
- conda-local-gen/restore
image
image
- conda build/restore
image
image

## Modules
- [x] Client

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/utils/venv_pack.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    from .venv import check_valid_venv_prefix
+
+    if not check_valid_venv_prefix(prefix):
+        raise FormatError(f"venv prefix: {prefix}")
+
+    # TODO: check editable packages
+    temp_dir = tempfile.mkdtemp(prefix="starwhale-venv-", dir=os.path.dirname(output))
+
+    try:
+        copy_fs(str(prefix), temp_dir)
```
Comment (jialeicui) [inline client/starwhale/utils/venv_pack.py:26]: Could you please add some comments for the reason copying files to a temp dir

File: client/starwhale/utils/venv_pack.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        with tarfile.open(output, "w:gz") as tar:
+            tar.add(str(temp_dir), arcname="")
```
Comment (jialeicui) [inline client/starwhale/utils/venv_pack.py:29]: Will tar to temp file and make an atomic mv be better?

Comment (tianweidut) [reply client/starwhale/utils/venv_pack.py:29]: ok

File: client/starwhale/utils/venv_pack.py (hunk 1 above)
Comment (tianweidut) [reply client/starwhale/utils/venv_pack.py]: Previous points:
venv dir --> copy to another unused dir -> tar the unused dir.
Making tar.gz will cost a lot of time if it works on a user's venv dir that may be changed. tar.gz will be broken.

But when I test the according scene， it works well.  So the copy_fs will be removed.

**Review by jialeicui — APPROVED** [review]

**Pull request merged**